src/riot-rs-embassy-common/Cargo.toml

```diff
@@ -10,8 +10,8 @@ repository.workspace = true
 workspace = true
 
 [dependencies]
-defmt = { workspace = true, optional = true }
-fugit = { workspace = true, optional = true }
+defmt = { workspace = true }
```

Comment:
I think we'd like to be careful with the features we require – this one has [a lot of unresolved questions](https://github.com/rust-lang/rust/issues/41517).

I trust there are stable options of making it work (at worst with some code duplication or macros), but before I even go through the hassle of working those out, can we get to the point of why a defmt dependency is a problem in the first place?

Reply:
We can have a trait which optionally has `Format` as suptertrait.
like [this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=0809ba739f24fcfa10ce599b49050db4).

This could be in `riot_rs_debug::MaybeFormat`:

```

#[cfg(feature = "defmt")]
trait MaybeFormat : Format;
#[cfg(not(feature = "defmt"))]
trait MaybeFormat;

Reply:
Sure we can do that. It's just one more trait everyone using it has to touch. (And for a user it's easy to just add Format to the derive set; MaybeFormat would need an extra impl, or a proc macro to make it derivable).

Again: What's the issue with defmt?